common/src/IO/AseParser.cpp

```diff
@@ -491,10 +486,9 @@ namespace TrenchBroom {
                 bounds.add(std::begin(mesh.vertices), std::end(mesh.vertices));
 
                 const auto textureIndex = geomObject.materialIndex;
-                auto* texture = textureIndex < textures.size() ? textures[textureIndex] : nullptr;
+                const auto* texture = textureIndex < surface.skinCount() ? surface.skin(textureIndex) : nullptr;
                 if (texture == nullptr) {
                     logger.warn() << "Invalid material index " << textureIndex;
-                    texture = loadDefaultTexture(m_fs, logger, "").release();
```

Comment:
This was a memory leak in the original code. I have changed it now so that it stores the default texture as a skin, and references that for invalid texture indices. See f4eee079e.